test/python/utils/mitigation/test_meas.py

```diff
@@ -0,0 +1,707 @@
+
+        Returns:
+            qubits: a list of qubits according to the given pattern
+            weight: the weight of the pattern_type,
+                    equals to the number of qubits
+
+        Additional Information:
+            qr[i] exists if and only if the i-th bit in the binary
+            expression of
+            pattern_type equals 1
+        """
+        qubits = []
+        weight = 0
+        for i in range(nq):
+            pattern_bit = pattern_type & 1
+            pattern_type = pattern_type >> 1
+            if pattern_bit == 1:
+                qubits.append(i)
+                weight += 1
+        return qubits, weight
+
+    def generate_ideal_results(self, state_labels, weight):
+        """
+        Generate ideal equally distributed results
+
+        Args:
+            state_labels (list): a list of calibration state labels
+            weight (int): the number of qubits
+
+        Returns:
+            results_dict: a dictionary of equally distributed results
+            results_list: a list of equally distributed results
+
+        Additional Information:
+            for each state in state_labels:
+            result_dict[state] = #shots/len(state_labels)
+        """
+        results_dict = {}
+        results_list = [0] * (2 ** weight)
+        state_num = len(state_labels)
+        for state in state_labels:
+            shots_per_state = self.shots / state_num
+            results_dict[state] = shots_per_state
+            # converting state (binary) to an integer
+            place = int(state, 2)
+            results_list[place] = shots_per_state
+        return results_dict, results_list
+
+    def test_ideal_meas_cal(self):
+        """Test ideal execution, without noise."""
+        for nq in self.nq_list:
+            for pattern_type in range(1, 2 ** nq):
+
+                # Generate the quantum register according to the pattern
+                qubits, weight = self.choose_calibration(nq, pattern_type)
+
+                # Generate the calibration circuits
+                meas_calibs, state_labels = complete_meas_cal(qubit_list=qubits, circlabel="test")
+
+                # Perform an ideal execution on the generated circuits
+                backend = Aer.get_backend("qasm_simulator")
+                job = qiskit.execute(meas_calibs, backend=backend, shots=self.shots)
+                cal_results = job.result()
+
+                # Make a calibration matrix
+                meas_cal = CompleteMeasFitter(cal_results, state_labels, circlabel="test")
+
+                # Assert that the calibration matrix is equal to identity
+                IdentityMatrix = np.identity(2 ** weight)
+                self.assertListEqual(
+                    meas_cal.cal_matrix.tolist(),
+                    IdentityMatrix.tolist(),
+                    "Error: the calibration matrix is \
+                                     not equal to identity",
```

Comment:
```suggestion
                    "Error: the calibration matrix is not equal to identity",
```
or just delete it entirely, I suppose.